base/applications/cmdutils/taskkill/taskkill.c

```diff
@@ -237,69 +289,116 @@ static int send_close_messages(void)
             }
         }
 
+        // Find processes to kill
         if (is_numeric)
         {
-            DWORD pid = atoiW(task_list[i]);
-            struct pid_close_info info = { pid };
-
-            if (pid == self_pid)
-            {
-                taskkill_message(STRING_SELF_TERMINATION);
-                status_code = 1;
-                continue;
-            }
-
-            EnumWindows(pid_enum_proc, (LPARAM)&info);
-            if (info.found)
-                taskkill_message_printfW(STRING_CLOSE_PID_SEARCH, pid);
-            else
+            WCHAR ps_name[MAX_PATH] = { 0 };
+            if (get_process_name_from_pid(atoiW(task_list[i]), ps_name, MAX_PATH))
             {
-                taskkill_message_printfW(STRING_SEARCH_FAILED, task_list[i]);
-                status_code = 128;
+                pkill_list[pkill_size] = atoiW(task_list[i]);
```

Comment:
Also in need of a bounds check/realloc

Reply:
I think it's not necessary. Because that statement is not in loop and size of `pkill_list` is equal to size of `pid_list`, so out of bound never happen.

Reply:
so the number of processes on your system never changes?
What happens when a new process is started?

Reply:
`pkill_list` just contains processes which will be killed. It's not used to contain running processes. `pid_list` will collect running processes on system and matched processes will be moved from `pid_list` to `pkill_list` to kill.